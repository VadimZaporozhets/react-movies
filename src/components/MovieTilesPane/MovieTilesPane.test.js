import React from 'react';
import { MovieTilesPane } from './index';
import { shallow } from 'enzyme';
import { emptyImageUrl } from '../../constants';

describe('<MovieTilesPane />', () => {
    it('should render movie tile pane with movies when no error found', () => {
        const props = {
            title: 'Title',
            movies: [
                {
                    title: 'Shazam',
                    poster_path: emptyImageUrl,
                    id: '1',
                    genres: ['Action', 'Comedy'],
                    releaseYear: '2006'
                },
                {
                    title: 'Star Wars',
                    poster_path: emptyImageUrl,
                    id: '2',
                    genres: ['Action', 'Comedy'],
                    releaseYear: '2013'
                }
            ],
            error: ''
        };

        const wrapper = shallow(<MovieTilesPane {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render error when it appears', () => {
        const props = {
            title: 'Title',
            movies: [],
            error: 'error'
        };

        const wrapper = shallow(<MovieTilesPane {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
