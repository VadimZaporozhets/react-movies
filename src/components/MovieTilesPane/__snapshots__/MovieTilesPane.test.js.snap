// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieTilesPane /> should render error when it appears 1`] = `
<Fragment>
  <WithStyles(Typography)
    variant="h4"
  >
    error
  </WithStyles(Typography)>
</Fragment>
`;

exports[`<MovieTilesPane /> should render movie tile pane with movies when no error found 1`] = `
<Fragment>
  <WithStyles(Grid)
    container={true}
    spacing={40}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        variant="h4"
      >
        Title
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(MovieTileComponent)
      genres={
        Array [
          "Action",
          "Comedy",
        ]
      }
      id="1"
      key="1"
      poster_path="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22288%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20288%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_164edaf95ee%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A14pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_164edaf95ee%22%3E%3Crect%20width%3D%22288%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%2296.32500076293945%22%20y%3D%22118.8%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
      releaseYear="2006"
      title="Shazam"
    />
    <WithStyles(MovieTileComponent)
      genres={
        Array [
          "Action",
          "Comedy",
        ]
      }
      id="2"
      key="2"
      poster_path="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22288%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20288%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_164edaf95ee%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A14pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_164edaf95ee%22%3E%3Crect%20width%3D%22288%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%2296.32500076293945%22%20y%3D%22118.8%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
      releaseYear="2013"
      title="Star Wars"
    />
  </WithStyles(Grid)>
</Fragment>
`;
